### PR TITLE
[Placeholder] Replace createVariable with createPlaceholder in BackendTest.cpp

### PR DESCRIPTION
**Pull request by SplitInfinity**
**Description**
This commit replaces all calls to `createVariable` in `BackendTest.cpp`
with `createPlaceholder` as part of the migration from `Variables` to
`Placeholders`.

**Testing**
All unit tests, including `BackendTest.cpp`, still pass.
```
meghanl-mbp:build_DEBUG meghanl$ ./tests/backendTest 
[==========] Running 15 tests from 5 test cases.
[----------] Global test environment set-up.
[----------] 2 tests from Interpreter
[ RUN      ] Interpreter.NotImplementedSave
[       OK ] Interpreter.NotImplementedSave (3 ms)
[ RUN      ] Interpreter.profileQuantizationForANetwork
[       OK ] Interpreter.profileQuantizationForANetwork (3 ms)
[----------] 2 tests from Interpreter (6 ms total)

[----------] 1 test from Context
[ RUN      ] Context.basicContextTest
[       OK ] Context.basicContextTest (0 ms)
[----------] 1 test from Context (0 ms total)

[----------] 4 tests from Interpreter/BackendTest
[ RUN      ] Interpreter/BackendTest.simpleInference/0
[       OK ] Interpreter/BackendTest.simpleInference/0 (542 ms)
[ RUN      ] Interpreter/BackendTest.debugPrint/0
print: %input = WeightVar float<4> mutable
shape: ( 4 )
max: 3.000  min: 0.000
[0.000, 1.000, 2.000, 3.000, ]

[       OK ] Interpreter/BackendTest.debugPrint/0 (0 ms)
[ RUN      ] Interpreter/BackendTest.decoupleCodegenFromGraph/0
[       OK ] Interpreter/BackendTest.decoupleCodegenFromGraph/0 (0 ms)
[ RUN      ] Interpreter/BackendTest.simplePlaceholderValue/0
[       OK ] Interpreter/BackendTest.simplePlaceholderValue/0 (1 ms)
[----------] 4 tests from Interpreter/BackendTest (543 ms total)

[----------] 4 tests from JIT/BackendTest
[ RUN      ] JIT/BackendTest.simpleInference/0
[       OK ] JIT/BackendTest.simpleInference/0 (368 ms)
[ RUN      ] JIT/BackendTest.debugPrint/0
print
shape: ( 4 )
max: 3.000  min: 0.000
[0.000, 1.000, 2.000, 3.000, ]
[       OK ] JIT/BackendTest.debugPrint/0 (36 ms)
[ RUN      ] JIT/BackendTest.decoupleCodegenFromGraph/0
[       OK ] JIT/BackendTest.decoupleCodegenFromGraph/0 (25 ms)
[ RUN      ] JIT/BackendTest.simplePlaceholderValue/0
[       OK ] JIT/BackendTest.simplePlaceholderValue/0 (22 ms)
[----------] 4 tests from JIT/BackendTest (451 ms total)

[----------] 4 tests from OpenCL/BackendTest
[ RUN      ] OpenCL/BackendTest.simpleInference/0
[       OK ] OpenCL/BackendTest.simpleInference/0 (42 ms)
[ RUN      ] OpenCL/BackendTest.debugPrint/0
print: %input = WeightVar float<4> mutable
shape: ( 4 )
max: 3.000  min: 0.000
[0.000, 1.000, 2.000, 3.000, ]

[       OK ] OpenCL/BackendTest.debugPrint/0 (4 ms)
[ RUN      ] OpenCL/BackendTest.decoupleCodegenFromGraph/0
[       OK ] OpenCL/BackendTest.decoupleCodegenFromGraph/0 (5 ms)
[ RUN      ] OpenCL/BackendTest.simplePlaceholderValue/0
[       OK ] OpenCL/BackendTest.simplePlaceholderValue/0 (5 ms)
[----------] 4 tests from OpenCL/BackendTest (56 ms total)

[----------] Global test environment tear-down
[==========] 15 tests from 5 test cases ran. (1056 ms total)
[  PASSED  ] 15 tests.
]
```

### CODE REVIEW

File: tests/unittests/BackendTest.cpp [hunk 1]
```diff
@@ -103,15 +103,14 @@ TEST(Interpreter, profileQuantizationForANetwork) {
...
 TEST_P(BackendTest, simpleInference) {
   Tensor inputs(ElemKind::FloatTy, {1, 32, 32, 3});
-  Context ctx;
 
   auto &mod = EE_.getModule();
   Function *F = mod.createFunction("main");
   F->setName("interpret");
-  auto *input = mod.createVariable(ElemKind::FloatTy, {1, 32, 32, 3}, "input",
-                                   VisibilityKind::Public);
+  auto *input =
```
[the remark this answers is not in the document]
Comment (SplitInfinity) [reply tests/unittests/BackendTest.cpp]: The default value for `Variable` is `true`, which is why I set it like that.

[the remark this answers is not in the document]
Comment (rdzhabarov) [reply tests/unittests/BackendTest.cpp]: got it, I think it does not make sense to make it trainable.

[the remark this answers is not in the document]
Comment (SplitInfinity) [reply tests/unittests/BackendTest.cpp]: Fixed.

File: tests/unittests/BackendTest.cpp [hunk 2]
```diff
@@ -52,22 +53,22 @@ TEST(Interpreter, profileQuantizationForANetwork) {
   Tensor inputs(ElemKind::FloatTy, {1, 4});
   inputs.getHandle() = {1, 1.2f, 0.5f, 1.3f};
 
-  auto *A = mod.createVariable(ElemKind::FloatTy, {1, 4}, "A",
-                               VisibilityKind::Public);
-  auto *Ex = mod.createVariable(ElemKind::FloatTy, {1, 4}, "E",
-                                VisibilityKind::Public);
+  auto *A = mod.createPlaceholder(ElemKind::FloatTy, {1, 4}, "A", true);
```
Comment (rdzhabarov) [inline tests/unittests/BackendTest.cpp]: no need for making vars/placeholders trained in this whole file.

**Pull request merged**